logic/broadcaster.js

```diff
@@ -242,21 +247,21 @@ __private.filterQueue = function(cb) {
 };
 
 /**
- * Checks if transaction is in pool or confirm it.
+ * Checks if transaction of a given id is in pool or confirmed in database "trs" table.
  *
  * @private
- * @param {transaction} transaction
+ * @param {string} transactionId
  * @param {function} cb
  * @returns {SetImmediate} null, boolean
  * @todo Add description for the params
  */
-__private.filterTransaction = function(transaction, cb) {
-	if (transaction !== undefined) {
-		if (modules.transactions.transactionInPool(transaction.id)) {
+__private.filterTransaction = function(transactionId, cb) {
+	if (transactionId !== undefined) {
```

Comment:
no need for explicit check, we can use `if (transactionId) {` this will ensure both `null` and `undefined`.